Update documentation to allow combining SHOW and TERMINATE TRANSACTION commands (#1501)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

Terminate transaction now allows `YIELD`.
The `WHERE` clause is not allowed on its own, as it is for `SHOW`, but needs the `YIELD` clause.

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW TRANSACTIONS [transaction-id[,...]]
----
[source, cypher, role="noheader"]
----
TERMINATE TRANSACTIONS transaction-id[,...]
----
a|

> The `transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameter.

Reviewer: The `transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameters.

> The `transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameter.

Reviewer: ```suggestion
The `transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameters.
```

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW TRANSACTIONS [transaction-id[,...]]
YIELD field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
  [WHERE expression]
TERMINATE TRANSACTIONS transaction-id[,...]
YIELD field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
  [WHERE expression]
RETURN field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
----
a|
The `SHOW` and `TERMINATE TRANSACTIONS` commands can be combined in the same query.

> The order of the commands does not matter and there can be more than one of each command.

Reviewer: ```suggestion
The order of the commands does not interfere and there can be more than one of each command.
```

Author: Not sure if that is the correct word either. The result might differ depending on the order (and your privileges) but they can be used in any order. Not sure what word do fit 🤔

Reviewer: Hmmm right. So maybe "matter" could work or maybe a rephrasing like "The command does not require a specific order and there can be more than one of each command"?

Author: Maybe something like?
```suggestion
The query does not require a specific order and there can be zero or more of each command type.
```

Reviewer: The way it is written now, I understand that the query can have zero commands? Is that possible? If not, maybe it would make more sense to be like this? 

```suggestion
The query does not require a specific order and there can be more than one of each command.
```

Author: It can't have 0 total but it could have only SHOW for example

Author: This sentence is not easy 😬

Reviewer: I agree! 🥲 What about this? 

```suggestion
The query does not require a specific order or a minimum number of commands.
```

Or then completely remove the mention to the amount of commands, or is this important information that must be explicit?

```suggestion
The query does not require a specific order.
```

Author: I don't think it needs to be explicit but it felt more correct to have something about it. But your first sentence still looks like we allow 0 commands total which is wrong.
```suggestion
The query does not require a specific order and there can be zero or more of each command type, however at least one command is needed.
```

Reviewer: I guess now it's much better! Thanks for the suggestion!

> When the command is not in standalone mode, the `YIELD` and `RETURN` are mandatory and `YIELD *` is not allowed.

Reviewer: ```suggestion
When the command is not in standalone mode, the `YIELD` and `RETURN` are mandatory. 
`YIELD *` is not allowed.
```

> The `transaction-id` is a comma-separated list of one or more quoted strings or an expression resolving to a string or a list of strings (for example the output column from `SHOW`).

Reviewer: ```suggestion
The `transaction-id` is a comma-separated list of one or more quoted strings.
It could also be an expression resolving to a string or a list of strings (for example the output column from `SHOW`).
```

Reviewer: @Hunterness Hey Therese, thanks for the heads up. I'm making a few editorial suggestions that might improve the experience of the reader. Nothing too major. Hope it helps!

Terminate transaction now allows `YIELD`.
The `WHERE` clause is not allowed on its own, as it is for `SHOW`, but needs the `YIELD` clause.

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW TRANSACTIONS [transaction-id[,...]]
----
[source, cypher, role="noheader"]
----
TERMINATE TRANSACTIONS transaction-id[,...]
----
a|

> The `transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameters.

Reviewer: ```suggestion
The `transaction-id` clause now allows general expressions resolving to a string or a list of strings instead of just parameters.
```

Author: I would not say that the id is a clause though?

Reviewer: My only issue is that adding 'the' would require some reference to what transaction id is. But another way to do this is by simply removing "the" : 
```suggestion
`transaction-id` now allows general expressions resolving to a string or a list of strings instead of just parameters.
```

Author: it is The id :P But I can see what you mean

a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
SHOW TRANSACTIONS [transaction-id[,...]]
YIELD field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
  [WHERE expression]
TERMINATE TRANSACTIONS transaction-id[,...]
YIELD field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
  [WHERE expression]
RETURN field[, ...]
  [ORDER BY field[, ...]]
  [SKIP n]
  [LIMIT n]
----
a|
The `SHOW` and `TERMINATE TRANSACTIONS` commands can be combined in the same query.
The query does not require a specific order and there can be zero or more of each command type, however at least one command is needed.

> When the command is not in standalone mode, the `YIELD` and `RETURN` are mandatory. 

Reviewer: ```suggestion
When the command is not in standalone mode, the `YIELD` and `RETURN` clauses are mandatory. 
```

`YIELD *` is not allowed.

> The `transaction-id` is a comma-separated list of one or more quoted strings.

Reviewer: ```suggestion
The `transaction-id` clause is a comma-separated list of one or more quoted strings.
```

> The `transaction-id` is a comma-separated list of one or more quoted strings.

Reviewer: ```suggestion
`transaction-id` is a comma-separated list of one or more quoted strings.
```
